pkg/sbom/cyclonedx/marshal_test.go

```diff
@@ -224,14 +224,14 @@ func TestMarshaler_Marshal(t *testing.T) {
 				},
 				Components: &[]cdx.Component{
 					{
-						BOMRef:  "pkg:rpm/centos/binutils@2.30-93.el8?arch=aarch64&distro=centos-8.3.2011",
+						BOMRef:  "pkg:rpm/centos/binutils@2.30-93.el8?arch=aarch64&upstream=binutils&distro=centos-8.3.2011",
```

Comment:
The `upstream` reference works for DPKG packages.
For RPMs the `upstream` should point to the SRC package that is coming from e.g. `upstream=vim-8.2.2637-20.el9_1.src.rpm`

Reply:
Thanks for information. I will try to fix it.

Reply:
@beltran-rubo i updated logic.
Do you have time to check it out?

Reply:
It looks correct to me. For `apk` packages the {{purl}} includes the {{upstream}} with the name of the package that comes from only, same approach than Deb packages.

Reply:
Cool! Thanks for your help!